src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -0,0 +1,66 @@
+package cromwell.engine.workflow
+
+import cromwell.binding._
+import cromwell.engine.ExecutionStatus
+import cromwell.engine.workflow.WorkflowActor.ExecutionStore
+
+sealed trait ExecutionStoreKey {
+  val scope: Scope
+
+  val index: Option[Int]
+
+  val parent: Option[ExecutionStoreKey]
+}
+
+
+case class CallKey(override val scope: Call,
```

Comment:
the more i'm seeing these override vals the more I'm disliking it. One of the killer apps of case classes is their simplicity and now that's going away